### ISSUE TITLE
Handle edge case: when device token is changed but Ably client doesn't exist

### ISSUE THREAD

**Issue by ben-xD** [ably/ably-cocoa#1159]
In Ably-java, there is an ActivationContext with keeps a specific token (DeviceTokenDetails) so that it can update Ably servers with the latest FCM registration token whenever it it is expired/ renewed by Android runtime. Because it has this token, it doesn't need an Ably client (rest/ realtime) to currently exist. The only requirement is that an ably client has been instantiated at least once in the whole lifetime of the app since it was installed (assuming the DeviceTokenDetails doesn't expire). This is because the DeviceTokenDetails will be created saved locally for future use when the ably client doesn't exist yet. In ably-java, you just call
```
ActivationContext.getActivationContext(this).onNewRegistrationToken(RegistrationToken.Type.FCM, s);
```

However, in Ably-cocoa, there is no such thing that would update Ably with the latest APNs token when it is renewed. In Ably-cocoa, to provide Ably servers with the APNs device token, a realtime or rest client is **needed**, using `ARTPush.didRegisterForRemoteNotifications withDeviceToken: realtime:`. This is bad in 2 ways:
- It is tedious for developers: after registering for APNs, they must save the deviceToken (in a property) and remember to give it to Ably when they create a realtime/ rest client. It adds complexity when there may be more than 1 Ably client (which client do you use?, does it matter if you give it to both, etc.).
- It goes against the Apple documentation. According to [Configuring Remote Notification Support](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1), 

> Each time your app launches, it must register with APNs. The methods to use differ according to the platform, but in all cases it works as follows:
> 
> Your app asks to be registered with APNs.
> 
> On successful registration, APNs sends an app-specific device token to the device.
> 
> The system delivers the device to your app by calling a method in your app delegate.
> 
> Your app sends the device token to the app’s associated provider

If a new (and different) device token is provided to the device at app launch, Ably will not know about it until a new Ably client and the `[ARTPush didRegisterForRemoteNotificationsWithDeviceToken:plugin.didRegisterForRemoteNotificationsWithDeviceToken_deviceToken rest:rest]` method is called with the new APNs device token and the rest/ realtime client.

# TLDR

We do not handle a theoretical edge case: when the device token is changed on iOS but the Ably client has not been created. Theoretically, it can change. I think its debatable if we handle this github issue/ case. By making `ARTPush.didRegisterForRemoteNotifications` or similar method which does not require a client, it would be easier for users to implement. It will also be more consistent with ably-java.

**Comment by maratal** [comment]
I agree, we should free users from calling `ARTPush.didRegisterForRemoteNotifications(withDeviceToken:realtime:)` manually.
